include/tc/core/libraries.h

```diff
@@ -140,6 +140,7 @@ constexpr auto boundsAsTemplate = R"C(
 template<typename T> inline __device__ T floord(T n, T d) {
   return n < 0 ? - (-n + d - 1)/d : n / d;
 }
+#define if_then_else(cond,a,b) (cond) ? (a) : (b);
```

Comment:
Is this used anywhere?

Reply:
Halide generates a call to that if you use ?:, I am open to a better way of implementing it.

Reply:
Oh, I misunderstood the scope of this header. Thought it was used in the compiler.